src/arch/x86_64/CodeGen.zig

```diff
@@ -4410,16 +4451,16 @@ fn airCondBr(self: *Self, inst: Air.Inst.Index) !void {
     // Capture the state of register and stack allocation state so that we can revert to it.
     const saved_state = try self.captureState();
 
-    try self.branch_stack.append(.{});
-    errdefer {
-        _ = self.branch_stack.pop();
-    }
+    {
```

Comment:
Why do we need explicit blocks here?

Reply:
The `errdefer` _must_ not trigger once the `pop` is executed in the normal code path.

Reply:
Correct, and then the block ends killing the `errdefer`, and _then_ we are allowed to perform the `pop` normally and proceed with more `try` expressions.

Reply:
Oh I see, right, good catch!